### PR TITLE
docs: clarify pattern usage with commands

### PR DESCRIPTION
**Pull request by remyabel**
Fixes: #5176

### CODE REVIEW

File: docs/usage/help.rst.inc [hunk 1]
```diff
@@ -23,7 +23,8 @@ store all files as `some/path/.../file.ext` and ``borg create
 File patterns support these styles: fnmatch, shell, regular expressions,
 path prefixes and path full-matches. By default, fnmatch is used for
 ``--exclude`` patterns and shell-style is used for the experimental
-``--pattern`` option.
+``--pattern`` option. For commands that support patterns in their ``PATH``
+argument like (``borg list``), the default pattern is path prefix.
```
Comment (ThomasWaldmann) [inline docs/usage/help.rst.inc:27]: see 1st line of this file: this is autogenerated.

so you need to revert this change and do the respective change in `archiver.py`.

File: docs/usage/list.rst [hunk 2]
```diff
@@ -32,3 +32,12 @@ Examples
...
     -rw-rw-r-- user   user    1416192 Sun, 2015-02-01 11:00:00 code/myproject/file.ext
     ...
+
+    $ borg list /path/to/repo::archiveA 'code/myproject'
+    drwxrwxr-x user   user          0 Sun, 2015-02-01 11:00:00 code/myproject
+    -rw-rw-r-- user   user    1416192 Sun, 2015-02-01 11:00:00 code/myproject/file.ext
+    ...
+
+    $ borg list /path/to/repo::archiveA 'pf:code/myproject'
+    drwxrwxr-x user   user          0 Sun, 2015-02-01 11:00:00 code/myproject
```
Comment (ThomasWaldmann) [inline docs/usage/list.rst:42]: pf is maybe not the best example here - who would use this in practice like that?

maybe rather `re:....`?

Comment (ThomasWaldmann) [reply docs/usage/list.rst:42]: also, please try whether this actually works. :-)

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
thanks for the PR, found some issues...

File: docs/usage/help.rst.inc (hunk 1 above)
Comment (remyabel) [reply docs/usage/help.rst.inc]: Sorry, forgot.

File: docs/usage/list.rst (hunk 2 above)
Comment (remyabel) [reply docs/usage/list.rst]: I tried the pattern on my existing repo. Let me know if it's still wrong.

File: docs/usage/list.rst [hunk 3]
```diff
@@ -32,3 +32,7 @@ Examples
     drwxrwxr-x user   user          0 Sun, 2015-02-01 11:00:00 code/myproject
     -rw-rw-r-- user   user    1416192 Sun, 2015-02-01 11:00:00 code/myproject/file.ext
     ...
+
+    $ borg list /path/to/repo/::archiveA --pattern "re:ext$"
```
Comment (ThomasWaldmann) [inline docs/usage/list.rst:36]: `'re:\.ext$'` would make sense:

- shows how to escape a dot (please check whether \ or \\ is needed)
- uses single quotes, so the shell does not expand anything
- makes sure only .ext extension matches and not .text also

Comment (remyabel) [reply docs/usage/list.rst]: Should be good now. As far as I can tell both `\` and `\\` have the same effect.

**Review by ThomasWaldmann — APPROVED** [review]
LGTM
